docs/upgrade/canary-upgrade.md

```diff
@@ -0,0 +1,371 @@
+# Istio canary upgrade guide
+
+## High level flow
+
+Let us suppose that we have an Istio 1.6 control plane running in our cluster and we would like to upgrade to an Istio 1.7 control plane.
+Here are the high level steps necessary to accomplish this through the canary upgrade flow:
+
+1. Deploy an Istio 1.7 control plane *next* to the Istio 1.6 control plane
+1. Migrate the data plane *gradually* to the new control plane
+1. When migration is fully finished, uninstall the Istio 1.6 control plane
+
+![Istio control plane canary upgrade](istio-cp-canary-upgrade.gif)
+
+This canary upgrade model is much safer than the in-place upgrade workflow, mainly because it can be done gradually, at your own pace.
+If issues arise they will probably only affect a small portion of your applications, initially, and, if that does happen, it should be much easier to rollback only a part of your application for use with the original control plane.
```

Comment:
```suggestion
If issues arise they will only affect a small portion of your applications, that you have selected as the initial scope for the upgrade. If that does happen, it will be much easier to rollback only a part of your application for use with the original control plane.
```